src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java

```diff
@@ -111,8 +140,11 @@ public ResponseEntity<String> putUserCommonsByIdBuy(
         .orElseThrow(
             () -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
 
-        if(userCommons.getTotalWealth() >= commons.getCowPrice() ){
+        if(userCommons.getTotalWealth() >= commons.getCowPrice()){
           userCommons.setTotalWealth(userCommons.getTotalWealth() - commons.getCowPrice());
+          // NOTE: Uncomment below if newly purchased cows should have full 100 health.
+          // userCommons.setCowHealth((userCommons.getCowHealth() * userCommons.getNumOfCows() + 100)/(userCommons.getNumOfCows() + 1));
```

Comment:
Ok, so your formula here is correct, but I think it's not going to be immediately obvious to a reader why it is correct.

If newly purchased cows should have 100% health, it makes sense that the new average health should be a weighted average, like this:

```
newNumCows = oldNumCows + 1;
newHealth = oldHealth * (newNumCows - 1)/newNumCows + 100*(1/newNumCows)
```

So, if we do the algebra and group terms, that reduces to:

newNumCows = oldNumCows + 1;
newHealth = (oldHealth * (newNumCows - 1) + 100) /  (newNumCows)

Which, with a substitution of variables of "userCommons.getNumOfCows" for "oldNumCows" is exactly that you have.

It's just that its... hard to see at first glance.

I'm not going to ask that you change it; I'm just putting this here so that when we come back to this later, and ask... what?!?!   There is an explanation :-)

Reply:
Yeah, I simplified everything to save up on some operations (and reduces mutations on the operators)